packages/nimble-components/src/table/specs/README.md

```diff
@@ -0,0 +1,70 @@
+# Nimble Table High Level Design
```

Comment:
I envision a few more PRs covering design decisions like:
 - element API
 - performance testing
 - accessibility
 - everything else in our [custom component template](https://github.com/ni/nimble/blob/main/specs/templates/custom-component.md) (this might be split across several PRs)

I'm trying to envision the right document breakdown for those. 

1. Will all of it go in one mega spec, with this being the first set of content? If so then maybe we should format this document using the custom-component template.
2. Or will we have separate documents? If so then maybe we should call this `table-tech-stack.md`

I'm also OK to defer this decision until we make our next design PR, and reformat/rename this document then if needed.